realm/realm-annotations-processor/src/test/resources/io/realm/some_test_EmbeddedClassSimpleParentRealmProxy.java

```diff
@@ -669,30 +669,16 @@ public static void insertOrUpdate(Realm realm, Iterator<? extends RealmModel> ob
 
             OsList childrenOsList = new OsList(table.getUncheckedRow(objKey), columnInfo.childrenColKey);
             RealmList<some.test.EmbeddedClass> childrenList = ((some_test_EmbeddedClassSimpleParentRealmProxyInterface) object).realmGet$children();
-            if (childrenList != null && childrenList.size() == childrenOsList.size()) {
-                // For lists of equal lengths, we need to set each element directly as clearing the receiver list can be wrong if the input and target list are the same.
-                int objectCount = childrenList.size();
-                for (int i = 0; i < objectCount; i++) {
-                    some.test.EmbeddedClass childrenItem = childrenList.get(i);
+            childrenOsList.removeAll();
```

Comment:
Has the generated proxy code changed here?

Reply:
Yes, the generated code for inserting single vs collection methods was different on embedded objects, and it shouldn't as it is the same operation.

I used the single implementation as it was the only one with a test, and actually, the generated code from the insert collection for embedded was not passing such test.